### PR TITLE
Implement `bootstrap-gstreamer` for all platforms

### LINKED ISSUE #25335: Support mach bootstrap-gstreamer on Windows
Currently installing gstreamer is a bit tedious on Windows:
Multiple msi's have to be installed, a package manager (chocolatey) has to be installed or pkg-conf-lite has to be installed manually and the paths have to be adjusted.

In my case, GStreamer was suggesting me E:\gstreamer as installation location, which is where pkg-config isn't looking. In fact it only seemed to look in it's own directory (provided servo doesn't pass a custom hardcoded path to where GStreamer usually lands, I suspect C:\gstreamer).

```bash
wget https://downloads.sourceforge.net/project/pkgconfiglite/0.28-1/pkg-config-lite-0.28-1_bin-win32.zip
unzip pkg-config-lite-0.28-1_bin-win32.zip -d pkg-config
cd pkg-config/pkg-config-lite-0.28-1/ && mv * ../ && cd ../ && rm -rf pkg-config-lite-0.28-1 && cd ../
```
This is the script I used to install pkg-config into `.servo/msvc-dependencies`, which would be Pull Request one: Adding this to mach as well as properly outputting an error when os != linux (and windows, so soon: when os == mac), because right now, `mach bootstrap-gstreamer` doesn't output anything when not on Linux.

This made servo build using
`PATH=$PATH:~/.cargo/bin:$(pwd)/.servo/msvc-dependencies/pkg-config/bin PKG_CONFIG_PATH="/e/gstreamer/1.0/x86_64/lib/pkgconfig" mach build -d`

So far so good, but then I hit 
```
 Finished release [optimized] target(s) in 39m 29s
Der Befehl "editbin" ist entweder falsch geschrieben oder
konnte nicht gefunden werden.
Packaging EGL DLLs
Packaging gstreamer DLLs
Could not find GStreamer installation directory.
Packaging MSVC DLLs
DLL file `api-ms-win-crt-runtime-l1-1-0.dll` not found!
Build FAILED in 0:39:29
```
Which is something mach does, I have to look into where it tries to locate GStreamer and why and how we could do that when installing gstreamer into msvc-dependencies (Or look in the PKG_CONFIG_PATH)

### PR DESCRIPTION
**Pull request by mrobinson**
This change makes it so that the Platform classes can now handle installing GStreamer dependencies and properly setting up the environment including when cross-compiling. For Windows and Linux, prepackaged GStreamer is now installed into `target/dependencies/gstreamer` when not installed system-wide. In addition this change:

1. The Environment path append helper is moved to `util.py` and a new `prepend` version is added.
2. `set_run_env` and `build_dev` functions are combined and include more code from callers so environment setup is more often in the same place. Now code that used to call `set_run_env` calls `build_dev` and then `os.environ.update(...)`. We can further refine this with the `is_build` argument later.
4. Python typing information is added in many places.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #25335
- [x] These changes do not require tests because they are to the build scripts.

### CODE REVIEW

File: python/servo/platform/macos.py [hunk 1]
```diff
@@ -9,28 +9,76 @@
...
+        env = os.environ.copy()
+        self.set_gstreamer_environment_variables_if_necessary(
+            env, cross_compilation_target, check_installation=False
+        )
+        gst_lib_dir = subprocess.check_output(
+            ["pkg-config", "--variable=libdir", "gstreamer-1.0"], env=env
+        )
+        if not gst_lib_dir.startswith(bytes(GSTREAMER_ROOT, "utf-8")):
+            print("GStreamer is installed, but not the official packages.\n"
+                  "Run `./mach bootstrap` or install packages from "
```
Comment (mukilan) [inline python/servo/platform/macos.py:52]: Should this be `./mach bootstrap-gstreamer`?

**Review by mukilan — APPROVED** [review]
Great work!

**Comment by mrobinson** [comment]
@bors-servo r=mukilan

Thanks for the review! I've addressed the comments above.

**Comment by mrobinson** [comment]
@bors-servo retry

This failed due to network issues on the GitHub runner.

**Pull request merged**